final-project/scripts/main.js

```diff
@@ -8,3 +8,156 @@ const after = before
+  let itemClass = itemC
+  let section = htmlSection
+
+  const loadCarrouselData = () => { 
+    let oReq = new XMLHttpRequest();
+    oReq.addEventListener("progress", updateProgress);
+    oReq.addEventListener("load", transferComplete);
+    oReq.addEventListener("error", transferFailed);
+    oReq.addEventListener("abort", transferCanceled);
+  
+    oReq.open("GET", theURL);
+    oReq.send();
+
+    function updateProgress (oEvent) {
+      if (oEvent.lengthComputable) {
+        var percentComplete = oEvent.loaded / oEvent.total * 100;
+      } else {
+        // Unable to compute progress information since the total size is unknown
+      }
+    } 
+
+    function transferComplete(evt){
+      let dataFromServer = this.responseText;
+      generateCarrouselItem(dataFromServer);
+    }
+
+    function transferFailed(evt){
+      console.log("An error occurred while transferring the file.");
+    }
+    
+    function transferCanceled (evt){
+      console.log("The transfer has been canceled by the user.");
+    }
+
+  }
+
+  const generateCarrouselItem = (items) => {
+    let jsonItem = JSON.parse(items);
+    activatePager(jsonItem.articles.length);
+    jsonItem.articles.forEach(element => {
+      let htmlTemplate = htmlPart(section,element);
+      let carrousel= selected;
+      let carrouselContent = carrousel.querySelector('.carrousel__content');
+      let asNode =  document.createElement(itemTag);
+      asNode.innerHTML =  htmlTemplate;
+      asNode.classList= itemClass + " carrousel__content__item";
+      carrouselContent.appendChild(asNode);
+    });
+  }
+  
+  const currentPage = (page) => {
+    let pageNum = page.target.dataset.index;
+    let carrousel = selected;
+    let carrouselContent = carrousel.querySelector('.carrousel__content');
+    let currentlyActive = carrousel.querySelector('.carrousel__controls .page-controller__indicator--active');
+    if (currentlyActive) {
+      currentlyActive.classList.toggle('page-controller__indicator--active');
+    }
+    let carrouselVisor = carrousel.querySelector('.carrousel__visor');
+    carrouselContent.style.left= (carrouselVisor.clientWidth*pageNum)*-1 +'px';
+    page.target.classList.toggle('page-controller__indicator--active');
+  }
+  
+  const activatePager = (numPages)  => {
+    let carrousel= selected;
+    let carrouselControls = carrousel.querySelector('.carrousel__controls');
+    for (let i = 0; i < numPages; i++) {
+      let pageNode =  document.createElement('span');
+      let addClass = i == 0 ? 'page-controller__indicator page-controller__indicator--active' : 'page-controller__indicator ';
+      pageNode.classList= addClass ;
+      pageNode.addEventListener("click", currentPage); 
+      pageNode.dataset.index = i;
+      carrouselControls.appendChild(pageNode);
+    }
+  }
+
+  const htmlPart = (section,element) => {
+    switch (section) {
+      case "blog":
+        return `
+        <div class="blog__content__article__image">
```

Comment:
Probably it would have been more readable and maintainable to have these divs cloned and transformed from the HTML instead of creating them from scratch with JS.